### PR TITLE
Add onHover for locally defined variables

### PR DESCRIPTION
**Pull request by JordanBoltonMN**
* Pulls in the latest parser changes
* Refactored providers into their own folder
* Refactored analysis into its own folder
* Adds onHover behavior for locally defined variables (see the test file `hover.ts`)

### CODE REVIEW

File: src/powerquery-language-services/lineTokenAtPosition/lineTokenAtPosition.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
```
Comment (mattmasson) [inline src/powerquery-language-services/lineTokenAtPosition/lineTokenAtPosition.ts:2]: double header

File: src/powerquery-language-services/lineTokenAtPosition/lineTokenAtPositionUtils.ts [hunk 2]
```diff
@@ -1,17 +1,28 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
+// Copyright (c) Microsoft Corporation.
```
Comment (mattmasson) [inline src/powerquery-language-services/lineTokenAtPosition/lineTokenAtPositionUtils.ts:4]: double header

**Review by mattmasson — APPROVED** [review]

File: src/powerquery-language-services/lineTokenAtPosition/lineTokenAtPosition.ts (hunk 1 above)
Comment (JordanBoltonMN) [reply src/powerquery-language-services/lineTokenAtPosition/lineTokenAtPosition.ts:2]: Fixed

File: src/powerquery-language-services/lineTokenAtPosition/lineTokenAtPositionUtils.ts (hunk 2 above)
Comment (JordanBoltonMN) [reply src/powerquery-language-services/lineTokenAtPosition/lineTokenAtPositionUtils.ts]: Fixed.

**Pull request merged**